src/main/java/net/clgd/ccemux/rendering/tror/TRoRRenderer.java

```diff
@@ -0,0 +1,217 @@
+package net.clgd.ccemux.rendering.tror;
+
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.UnsupportedEncodingException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingDeque;
+
+import net.clgd.ccemux.emulation.EmulatedComputer;
+import net.clgd.ccemux.emulation.EmulatedPalette;
+import net.clgd.ccemux.emulation.EmulatedTerminal;
+import net.clgd.ccemux.rendering.Renderer;
+import net.clgd.ccemux.rendering.RendererConfig;
+
+public class TRoRRenderer implements Renderer, EmulatedTerminal.Listener, EmulatedPalette.Listener {
+	private static final String[] COLOURS = new String[]{
+			"0", "1", "2", "3", "4", "5", "6", "7",
+			"8", "9", "a", "b", "c", "d", "e", "f"
+	};
+
+	private final EmulatedComputer computer;
+	private final RendererConfig config;
+
+	private final List<Listener> listeners = new ArrayList<>();
+
+	private final BlockingDeque<InputProvider.InputPacket> events;
+
+	private final Writer output;
+
+	public TRoRRenderer(EmulatedComputer computer, RendererConfig config) {
+		this.computer = computer;
+		this.config = config;
+
+		try {
+			output = new OutputStreamWriter(System.out, "UTF-8");
+		} catch (UnsupportedEncodingException e) {
+			throw new RuntimeException(e);
+		}
+
+		computer.terminal.addListener(this);
+		computer.terminal.getEmulatedPalette().addListener(this);
+
+		events = InputProvider.getStdinProvider().getQueue(computer);
+	}
+
+	@Override
+	public void addListener(Renderer.Listener listener) {
+		listeners.add(listener);
+	}
+
+	@Override
+	public void removeListener(Renderer.Listener listener) {
+		listeners.remove(listener);
+	}
+
+	@Override
+	public boolean isVisible() {
+		return false;
+	}
+
+	@Override
+	public void setVisible(boolean visible) {
+
+	}
```

Comment:
I'd argue that `isVisible` should return `true` whenever the renderer is outputting terminal events - i.e. always.